agent/rpc/middleware/interceptors.go

```diff
@@ -33,33 +33,79 @@ var OneTwelveRPCSummary = []prometheus.SummaryDefinition{
 }
 
 type RequestRecorder struct {
-	Logger       hclog.Logger
-	RecorderFunc func(key []string, val float32, labels []metrics.Label)
+	Logger         hclog.Logger
+	RecorderFunc   func(key []string, val float32, labels []metrics.Label)
+	serverIsLeader func() bool
+	localDC        string
 }
 
-func NewRequestRecorder(logger hclog.Logger) *RequestRecorder {
-	return &RequestRecorder{Logger: logger, RecorderFunc: metrics.AddSampleWithLabels}
+func NewRequestRecorder(logger hclog.Logger, isLeader func() bool, localDC string) *RequestRecorder {
+	return &RequestRecorder{
+		Logger:         logger,
+		RecorderFunc:   metrics.AddSampleWithLabels,
+		serverIsLeader: isLeader,
+		localDC:        localDC,
+	}
 }
 
 func (r *RequestRecorder) Record(requestName string, rpcType string, start time.Time, request interface{}, respErrored bool) {
 	elapsed := time.Since(start).Milliseconds()
 	reqType := requestType(request)
+	serverRole := r.getServerRole()
 
 	labels := []metrics.Label{
 		{Name: "method", Value: requestName},
 		{Name: "errored", Value: strconv.FormatBool(respErrored)},
 		{Name: "request_type", Value: reqType},
 		{Name: "rpc_type", Value: rpcType},
+		{Name: "server_role", Value: serverRole},
 	}
 
+	labels = r.addOptionalLabels(request, labels)
+
 	// math.MaxInt64 < math.MaxFloat32 is true so we should be good!
 	r.RecorderFunc(metricRPCRequest, float32(elapsed), labels)
-	r.Logger.Trace(requestLogName,
-		"method", requestName,
-		"errored", respErrored,
-		"request_type", reqType,
-		"rpc_type", rpcType,
-		"elapsed", elapsed)
+
+	labelsArr := flattenLabels(labels)
+	r.Logger.Trace(requestLogName, labelsArr...)
```

Comment:
i did my best to not add another helper func here -- let me know if there's a easier method to enumerate keys, values as `[]interface{}` in go 🙈